x/ibc/20-transfer/module.go

```diff
@@ -257,5 +317,19 @@ func (am AppModule) OnTimeoutPacket(
 	if err := types.ModuleCdc.UnmarshalBinaryBare(packet.GetData(), &data); err != nil {
 		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "cannot unmarshal ICS-20 transfer packet data: %s", err.Error())
 	}
-	return handleTimeoutDataTransfer(ctx, am.keeper, packet, data)
+	// refund tokens
+	if err := am.keeper.OnTimeoutPacket(ctx, packet, data); err != nil {
+		return nil, err
+	}
+
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
```

Comment:
This event type is only for msg handlers afaik. This should instead be something like `EventTimeout`